www/icapeg/src/opnsense/service/templates/OPNsense/ICAPeg/config.toml

```diff
@@ -0,0 +1,142 @@
+resp_mode = true
+{% else %}
+resp_mode = false
+{% endif %}
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.shadow_service') and OPNsense.icapeg.clhashlookup.shadow_service == '1' %}
+shadow_service = true
+{% else %}
+shadow_service = false
+{% endif %}
+preview_bytes = "{{ OPNsense.icapeg.clhashlookup.preview_bytes }}"
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.preview_enabled') and OPNsense.icapeg.clhashlookup.preview_enabled == '1' %}
+preview_enabled = true
+{% else %}
+preview_enabled = false
+{% endif %}
+bypass_extensions = ["*"]
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.process_extensions') and OPNsense.icapeg.clhashlookup.process_extensions != '' %}
+process_extensions = [{% for item in OPNsense.icapeg.clhashlookup.process_extensions.split(',') %}"{{ item|trim }}"{{ ", " if not loop.last else "" }}{% endfor %}]
+{% endif %}
+reject_extensions = []
+scan_url = "{{ OPNsense.icapeg.clhashlookup.scan_url }}"
+timeout = {{ OPNsense.icapeg.clhashlookup.timeout }}
+fail_threshold = {{ OPNsense.icapeg.clhashlookup.fail_threshold }}
+max_filesize = {{ OPNsense.icapeg.clhashlookup.max_filesize }}
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.return_original_if_max_file_size_exceeded') and OPNsense.icapeg.clhashlookup.return_original_if_max_file_size_exceeded == '1' %}
+return_original_if_max_file_size_exceeded = true
+{% else %}
+return_original_if_max_file_size_exceeded = false
+{% endif %}
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.return_400_if_file_ext_rejected') and OPNsense.icapeg.clhashlookup.return_400_if_file_ext_rejected == '1' %}
+return_400_if_file_ext_rejected = true
+{% else %}
+return_400_if_file_ext_rejected = false
+{% endif %}
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.verify_server_cert') and OPNsense.icapeg.clhashlookup.verify_server_cert == '1' %}
+verify_server_cert = true
+{% else %}
+verify_server_cert = false
+{% endif %}
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.bypass_on_api_error') and OPNsense.icapeg.clhashlookup.bypass_on_api_error == '1' %}
+bypass_on_api_error = true
+{% else %}
+bypass_on_api_error = false
+{% endif %}
+http_exception_response_code = {{ OPNsense.icapeg.clhashlookup.http_exception_response_code }}
+{% if helpers.exists('OPNsense.icapeg.clhashlookup.http_exception_has_body') and OPNsense.icapeg.clhashlookup.http_exception_has_body == '1' %}
+http_exception_has_body = true
+{% else %}
+http_exception_has_body = false
+{% endif %}
+exception_page = {{ OPNsense.icapeg.clhashlookup.exception_page }}
+
+[clamav]
+vendor = "clamav"
+service_caption = "clamav service"
+service_tag = "CLAMAV ICAP"
+{% if helpers.exists('OPNsense.icapeg.clamav.req_mode') and OPNsense.icapeg.clamav.req_mode == '1' %}
+req_mode = true
+{% else %}
+req_mode = false
+{% endif %}
+{% if helpers.exists('OPNsense.icapeg.clamav.resp_mode') and OPNsense.icapeg.clamav.resp_mode == '1' %}
```

Comment:
`mode`?
mod stands for modification. Maybe a bit strange in this server...

Reply:
In the [ICAP RFC](https://datatracker.ietf.org/doc/rfc3507/) it's named `Request Modification Mode` see section 4.8.

Reply:
Ok, I only know it by the verbs - REQMOD and RESPMOD, not the actual headings in the RFC :).
resp_mod_mode reads even worse.

If fichtner is fine with it, then it is good. Keep in mind that if you use it on OPNsense, that the terms might be a good idea to align with the Squid config.

Reply:
@fabianfrz Got you. I'll try to keep it as close to Squid as possible. It's a variable in the config upstream so it will be hard to edit in the plugin. Thanks for the advice.

@fichtner Can you please give your opinion.